### PR TITLE
Add `docker.` prefix to container logger names, to enable log level to be set across all loggers

### PR DESCRIPTION
**Pull request by vcvitaly**
Fixes https://github.com/testcontainers/testcontainers-java/issues/3016

### CODE REVIEW

**Comment by vcvitaly** [comment]
@bsideup please review?

**Comment by hmemcpy** [comment]
Sorry for the bump, any chance of reviewing this? I have a feeling this may help deal with #637 where it would be nice to set the log level for all dockers at once instead of matching against the 🐳...

**Comment by rnorth** [comment]
@vcvitaly please could you merge from master, or preferably [give maintainers permission to edit](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)?

We've had some CI problems, so hopefully updating from master should fix things.

**Comment by rnorth** [comment]
Overall I think this looks fine and long overdue (sorry for the immense time to review). Let's just see the tests pass once rebased onto master and I think we should be good to go!

**Comment by rnorth** [comment]
I think it would be fair to say this fixes #637, although not in the way originally proposed.

**Comment by vcvitaly** [comment]
@rnorth I just can't find that checkbox to allow edits, I don't have it for some reason. I rebased onto the master.

**Review by rnorth — DISMISSED** [review]
Thanks @vcvitaly, LGTM.
Heads up @bsideup, @kiview, this is a slight change in aesthetics, but I think is unavoidable!

**Comment by rnorth** [comment]
I’m afraid I’ll have to defer this to 1.15.1, so that we can review this together. We’re expecting to publish that release soon after 1.15.0, so it shouldn’t be too much delay.

**Comment by perlun** [comment]
@rnorth Anything blocking this from being merged now, given that 1.15.1 is released? Would be incredibly nice to make the container logging be more easily controlled from e.g. `logback-test.xml` config.

**Comment by rnorth** [comment]
@bsideup I think you had some ideas about how this could be done without a change to how it looks for normal use cases?

File: core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java
Comment (adutra) [inline core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java:21]: I'd suggest avoiding brackets here. Some frameworks (e.g. Quarkus) give brackets a special meaning, and as a consequence, the following configuration would fail:

```properties
quarkus.log.category."docker[foo:latest]".level=ERROR
```

Of course, this would work:

```
quarkus.log.category.docker.level=ERROR
```

But then we lose the ability to distinguish different containers.

I think a better approach would be to create a hierarchy of `docker.<image name>.<label>`, separated by dots. 

Also, IMHO the little whale emoji is pretty, but only adds clutter when included in a logger name. Maybe move it to the log messages themselves?

**Comment by perlun** [comment]
Ping @bsideup, would be great to get your feedback on this. (Not myself advocating a particular solution, only eager to see #637 resolved one way or the other)

**Comment by okorz001** [comment]
any progress on this? it is a bit more difficult to promote this library when logging is difficult to configure. 

(personally, I would prefer an org.testcontainers. prefix, but anything other than root will solve my problems.)

**Comment by sideeffffect** [comment]
Humble ping :slightly_smiling_face:

**Comment by kiview** [comment]
Sorry, everyone, I am not sure why we let this go stale for some time now. We will look into it so that we can decide at least whether we will tackle this in a timely manner, or not at all.

**Comment by perlun** [comment]
> Sorry, everyone, I am not sure why we let this go stale for some time now. We will look into it so that we can decide at least whether we will tackle this in a timely manner, or not at all.

Sounds good @kiview, any updates on this yet? :relaxed:

File: core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java
Comment (perlun) [reply core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java:21]: I agree @adutra. The whale emoji is cute but it totally doesn't belong in the actual logger _name_. :joy: So something like this would be preferable here:

```java
        return LoggerFactory.getLogger("docker." + abbreviatedName);
```

Ideally, I think it should be prefixed with `org.testcontainers` though, so that all Testcontainers-based logging can be filtered away easily. `org.testcontainers.docker` would perhaps be the most suitable prefix here, i.e. this:

```java
        return LoggerFactory.getLogger("org.testcontainers.docker." + abbreviatedName);
```

**Comment by perlun** [comment]
@vcvitaly - I added some feedback in https://github.com/testcontainers/testcontainers-java/pull/3018/files#r1006459726. If you're still with us, it would be nice to get these final details sorted out before this hopefully gets merged.

If not, I am willing to take ownership of this change to make it be mergable. We really, really need to get this shipped anytime soon. It's a major pain related to e.g. CI logging right now.

**Comment by vcvitaly** [comment]
> @vcvitaly - I added some feedback in https://github.com/testcontainers/testcontainers-java/pull/3018/files#r1006459726. If you're still with us, it would be nice to get these final details sorted out before this hopefully gets merged.
> 
> If not, I am willing to take ownership of this change to make it be mergable. We really, really need to get this shipped anytime soon. It's a major pain related to e.g. CI logging right now.

@perlun sorry, I almost forgot about this) let me review this within two days

**Comment by vcvitaly** [comment]
@kiview I renamed the logger as suggested by the community, please provide your feedback.

**Comment by perlun** [comment]
Ping @kiview, is this mergable now or would it require any further adjustments?

**Comment by perlun** [comment]
Anyone who talks to the maintainers recently, please ping them about this. I seem to be unable to get in touch with them despite repeated attempts.

**Comment by eddumelendez** [comment]
Hi, sorry for the delay on this one. We just started discussing about this internally. So, we will be sharing news hopefully soon.

Thanks for your patience.

**Comment by vcvitaly** [comment]
@eddumelendez quite some time has passed since the PR was opened so here is a short review of the reasons why I opened it:
1. The initial problem was that since the container logs start with the whale emoji or docker prefix, they go under the root INFO hierarchy, and you cannot see debug logs that might be useful if there are issues. I experienced that myself when I was working on another PR to testcontainers and couldn't find an issue for a while because it was printed as debug and effectively not shown. What I did - I just removed the whale emoji from the factory and added the 'docker' prefix with the DEBUG level to the logback-test.xml because I thought that missing debug info is a mistake. I now understand that it was probably a conscious decision.
2. Since then multiple community members have made complaints that it's hard to configure logging with the whale emoji and that a non-Unicode prefix should be used. I then changed the prefix to org.testcontainers.docker as suggested by @perlun 

If something else is required from my side that might speed up pr review, I'd be happy to provide any additional info.

File: core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java
Comment (eddumelendez) [inline core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java:19]:
```suggestion
        return LoggerFactory.getLogger("tc." + abbreviatedName);
```

File: core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java
Comment (eddumelendez) [inline core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java:32]:
```suggestion
        assertThat(event.getLoggerName()).startsWith("tc");
```

File: modules/orientdb/src/test/resources/logback-test.xml
Comment (eddumelendez) [inline modules/orientdb/src/test/resources/logback-test.xml:18]: can we rollback this, please?

**Review by eddumelendez — CHANGES_REQUESTED** [review]
Thanks for your contribution, @vcvitaly ! Sorry about the delay. I've left some comments. Can you also update the [docs](https://github.com/testcontainers/testcontainers-java/blob/main/docs/supported_docker_environment/logging_config.md?plain=1#L19) related to logging, please?

The reason why `tc` has been decided it is because we want to avoid log truncation.

File: core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java
Comment (eddumelendez) [inline core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java:17]:
```suggestion
```

File: core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java
Comment (eddumelendez) [inline core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java:22]:
```suggestion
```

File: core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java
Comment (eddumelendez) [inline core/src/test/java/org/testcontainers/utility/DockerLoggerFactoryTest.java:25]:
```suggestion
```

**Comment by eddumelendez** [comment]
#6781 has been created in order to tackle this. Thanks for your contribution @vcvitaly

**Pull request closed without merging**
